### PR TITLE
mds: dump formatter even for errors

### PR DESCRIPTION
**Pull request by batrick**
The admin_socket framework only dumps the formatter by default if the command succeeds.

Fixes: a4dc88140071f9cdfe5654402067434728fee469
Fixes: https://tracker.ceph.com/issues/66184

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
asok_command always prepends and "ERROR" message so we just can't/shouldn't use it for QA testing.

**Comment by batrick** [comment]
(still more to fix...)

File: src/mds/MDSRank.cc [hunk 1]
```diff
@@ -2675,6 +2675,7 @@ void MDSRankDispatcher::handle_asok_command(
       f->close_section();
 
       bufferlist outbl;
+      f->flush(outbl); /* even for errors, dump f */
       do_respond(rc, {}, outbl);
```
Comment (leonid-s-usov) [inline src/mds/MDSRank.cc:2679]: I just realized that it would be nice to pass the CSS here

```suggestion
do_respond(rc, css, outbl);
```

**Review by leonid-s-usov — APPROVED** [review]

File: src/mds/MDSRank.cc (hunk 1 above)
Comment (batrick) [reply src/mds/MDSRank.cc]: done, PTAL

Comment (leonid-s-usov) [reply src/mds/MDSRank.cc]: ❤️

**Review by leonid-s-usov — CHANGES_REQUESTED** [review]
Please note that `tell_command` will assert for non-zero status. You should use `check_status=False` kwarg if you want to process the rc externally, the `asok_command` had it

**Review by leonid-s-usov — APPROVED** [review]
Looks good!

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
jenkins retest this please

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66261](https://tracker.ceph.com/issues/66261).

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/66261#note-3

**Comment by batrick** [comment]
Pulled vstart_runner change out as it was failing api tests. It's now at https://github.com/ceph/ceph/pull/57881

**Pull request merged**